snuba/query/snql/__init__.py

```diff
@@ -1,28 +1,296 @@
+        self, node: Node, visited_children: Tuple[Any, Any, Sequence[OrderBy], Any]
+    ) -> Sequence[OrderBy]:
+        _, _, order_columns, _ = visited_children
+        return order_columns
+
+    def visit_order_list(
+        self, node: Node, visited_children: Tuple[OrderBy, Expression, Node]
+    ) -> Sequence[OrderBy]:
+        left_order_list, right_order, order = visited_children
+        ret: List[OrderBy] = []
+
+        # in the case of one OrderBy
+        # left_order_list will be an empty node
+        if not isinstance(left_order_list, Node):
+            if not isinstance(left_order_list, (list, tuple)):
+                ret.append(left_order_list)
+            else:
+                for p in left_order_list:
+                    ret.append(p)
+
+        if order.text == "ASC":
+            ret.append(OrderBy(OrderByDirection.ASC, right_order))
+        elif order.text == "DESC":
+            ret.append(OrderBy(OrderByDirection.DESC, right_order))
+
+        return ret
+
+    def visit_order_columns(
+        self, node: Node, visited_children: Tuple[Expression, Node, Any, Any, Any]
+    ) -> OrderBy:
+        column, order, _, _, _ = visited_children
+
+        if order.text == "ASC":
+            order_object = OrderBy(OrderByDirection.ASC, column)
+        elif order.text == "DESC":
+            order_object = OrderBy(OrderByDirection.DESC, column)
+
+        return order_object
+
+    def visit_group_by_clause(
+        self, node: Node, visited_children: Tuple[Any, Any, Sequence[Expression], Any]
+    ) -> Sequence[Expression]:
+        _, _, group_columns, _ = visited_children
+        return group_columns
+
+    def visit_group_columns(
+        self, node: Node, visited_children: Tuple[Expression, Any, Any, Any]
+    ) -> Expression:
+        columns, _, _, _ = visited_children
+        return columns
+
+    def visit_group_list(
+        self, node: Node, visited_children: Tuple[Expression, Expression]
+    ) -> Sequence[Expression]:
+        left_group_list, right_group = visited_children
+        ret: List[Expression] = []
+
+        # in the case of one GroupBy / By
+        # left_group_list will be an empty node
+        if not isinstance(left_group_list, Node):
+            if not isinstance(left_group_list, (list, tuple)):
+                ret.append(left_group_list)
+            else:
+                for p in left_group_list:
+                    ret.append(p)
+
+        ret.append(right_group)
+        return ret
+
+    def visit_collect_clause(
+        self,
+        node: Node,
+        visited_children: Tuple[Any, Any, Sequence[SelectedExpression], Any],
+    ) -> Sequence[SelectedExpression]:
+        _, _, selected_columns, _ = visited_children
+        return selected_columns
+
+    def visit_selected_expression(
+        self, node: Node, visited_children: Tuple[Expression, Any]
+    ) -> SelectedExpression:
+        exp, _ = visited_children
+        return SelectedExpression(node.text.strip(), exp)
```

Comment:
Do you actually need strip() here ?

Reply:
Yes. otherwise there are trailing spaces. I think the reason for these spaces is the 'comma, space' format.